kie-soup-maven-utils/kie-soup-maven-integration/src/main/java/org/appformer/maven/integration/AbstractFilesArtifactResolver.java

```diff
@@ -99,7 +99,12 @@ private boolean isInJarStructuredFolder(String name, String type) {
     }
 
     private PomParser buildPomParser(AFReleaseId releaseId) {
-        List<URL> url = effectivePoms.stream().filter(e -> e.getFile().endsWith(toFile(releaseId, "pom"))).collect(toList());
+        final String pomName = toFile(releaseId, "pom");
+        List<URL> url = effectivePoms.stream().filter(e -> e.getFile().endsWith(pomName)).collect(toList());
+        if (url.isEmpty() && pomName.contains("-SNAPSHOT")) {
```

Comment:
find cannot be used directly over string, therefore you need the .* for matches to work, therefore 
```suggestion
       if (url.isEmpty() && releaseId.getVersion().endsWith("-SNAPSHOT")) {
            url =  effectivePoms.stream().filter(e -> e.getFile().matches(".*"+releaseId.getArtifactId()+"-"+releaseId.getVersion().replace("-SNAPSHOT",".*")+ ".pom$")).collect(toList());
        }
```

Reply:
Exactly I tried to make the simpler change to show the fix, Let's do this way to then